### PR TITLE
[WIP] non-comp queue

### PR DESCRIPTION
**Pull request by mdbenjam**
Resolves #{github issue number}

### Description
Please explain the changes you made here.

### Acceptance Criteria
- [ ] Code compiles correctly

### Testing Plan
1. Go to ...

### CODE REVIEW

File: app/controllers/appeals_controller.rb [hunk 1]
```diff
@@ -68,7 +68,8 @@ def show
   helper_method :appeal, :url_appeal_uuid
 
   def appeal
-    @appeal ||= Appeal.find_appeal_by_id_or_find_or_create_legacy_appeal_by_vacols_id(params[:appeal_id])
+    @appeal = Appeal.find_appeal_by_id_or_find_or_create_legacy_appeal_by_vacols_id(params[:appeal_id]) if params[:appeal_type] == "Appeal" || params[:appeal_type] == "LegacyAppeal"
```
Comment (mdbenjam) [inline app/controllers/appeals_controller.rb]: Need to abstract this somewhere, we'll need to send `appeal_type` everywhere.

File: app/models/claim_review.rb [hunk 2]
```diff
@@ -108,6 +108,10 @@ def find_request_issue_by_description(description)
     request_issues.find { |reqi| reqi.description == description }
   end
 
+  def external_id
```
Comment (mdbenjam) [inline app/models/claim_review.rb]: Thinking we might need a better name for external_id

File: client/app/queue/CaseDetailsView.jsx [hunk 3]
```diff
@@ -56,11 +58,16 @@ class CaseDetailsView extends React.PureComponent {
     const {
       appealId,
       appeal,
+      appealType,
       error,
       success,
       featureToggles
     } = this.props;
 
+    if (appealType === 'HigherLevelReview') {
```
Comment (mdbenjam) [inline client/app/queue/CaseDetailsView.jsx]: This component can become a wrapper for all of the "detailed versions"

File: client/app/queue/CaseSnapshot.jsx [hunk 4]
```diff
@@ -309,7 +309,7 @@ const mapStateToProps = (state: State, ownProps: Params) => {
   const { featureToggles, userRole, canEditAod } = state.ui;
 
   return {
-    appeal: appealWithDetailSelector(state, { appealId: ownProps.appealId }),
+    appeal: appealWithDetailSelector(state, { appealId: ownProps.appealId, appealType: ownProps.appealType }),
```
Comment (mdbenjam) [inline client/app/queue/CaseSnapshot.jsx]: We should be passing appeal down, not appealId/type and then going to the redux store. We should be able to make our components more reusable if they don't need to go to the store.

File: client/app/queue/HigherLevelReviewDetailsView.jsx [hunk 5]
```diff
@@ -0,0 +1,85 @@
+import { css } from 'glamor';
```
Comment (mdbenjam) [inline client/app/queue/HigherLevelReviewDetailsView.jsx]: This file would belong to Intake.

File: client/app/queue/NonCompOrganizationQueue.jsx [hunk 6]
```diff
@@ -0,0 +1,109 @@
+import React from 'react';
```
Comment (mdbenjam) [inline client/app/queue/NonCompOrganizationQueue.jsx]: This file would belong to Intake.

File: client/app/queue/OrganizationQueue.jsx [hunk 7]
```diff
@@ -26,12 +28,20 @@ const containerStyles = css({
   position: 'relative'
 });
 
+const NON_COMP_ORGANIZATIONS = ['nca'];
```
Comment (mdbenjam) [inline client/app/queue/OrganizationQueue.jsx]: Need to think of a better way to fork this.

File: client/app/queue/QueueActions.js [hunk 8]
```diff
@@ -19,24 +19,26 @@ import type {
 } from './types/models';
 
 export const onReceiveQueue = (
-  { tasks, amaTasks, appeals }:
+  { tasks, amaTasks, appeals, higherLevelReviews }:
```
Comment (mdbenjam) [inline client/app/queue/QueueActions.js]: Want to think of a more generic way to do this.

File: client/app/queue/QueueApp.jsx [hunk 9]
```diff
@@ -276,36 +281,36 @@ class QueueApp extends React.PureComponent<Props> {
             title="Unassigned Cases | Caseflow"
             render={this.routedJudgeQueueList('assign')} />
           <Route
-            path="/queue/appeals/:appealId/modal/advanced_on_docket_motion"
+            path="/queue/:appealType/:appealId/modal/advanced_on_docket_motion"
```
Comment (mdbenjam) [inline client/app/queue/QueueApp.jsx]: One of the big changes would be updating the URLs for the case details pages to be `queue/:appealType/:appealId`

File: client/app/queue/components/TaskTable.jsx [hunk 10]
```diff
@@ -272,8 +273,30 @@ class TaskTable extends React.PureComponent<Props> {
     } : null;
   }
 
+  claimantLinkColumn = () => {
```
Comment (mdbenjam) [inline client/app/queue/components/TaskTable.jsx]: New columns are defined here. Lowell has a good idea for a refactor here that will make it easier to place columns in an arbitrary order.

File: client/app/queue/reducers.js [hunk 11]
```diff
@@ -22,6 +22,7 @@ export const initialState = {
   amaTasks: {},
   appeals: {},
   appealDetails: {},
+  higherLevelReviews: {},
```
Comment (mdbenjam) [inline client/app/queue/reducers.js]: higher level reviews and supplement claims can be top level objects in our redux store.

File: client/app/queue/selectors.js [hunk 12]
```diff
@@ -58,8 +60,8 @@ export const getActiveModalType = createSelector(
 );
 
 export const tasksWithAppealSelector = createSelector(
-  [getTasks, getAmaTasks, getAppeals, getAppealDetails],
-  (tasks: Tasks, amaTasks: Tasks, appeals: BasicAppeals, appealDetails: AppealDetails) : Array<TaskWithAppeal> => {
+  [getTasks, getAmaTasks, getAppeals, getAppealDetails, getHigherLevelReviews],
```
Comment (mdbenjam) [inline client/app/queue/selectors.js]: Some of our selectors need to change to pull the necessary data.

File: client/app/queue/utils.js [hunk 13]
```diff
@@ -99,7 +99,32 @@ const extractAppealsFromTasks =
   (tasks: Array<Object>):
     BasicAppeals => {
     return tasks.reduce((accumulator, task) => {
-      if (!accumulator[task.attributes.external_appeal_id]) {
+      if (!accumulator[task.attributes.external_appeal_id] && (task.attributes.appeal_type === 'Appeal' || task.attributes.appeal_type === 'LegacyAppeal')) {
```
Comment (mdbenjam) [inline client/app/queue/utils.js]: Need to think of a more elegant way to do this.

**Comment by jiahuang** [comment]
@mdbenjam I'm looking into what it takes to do the frontend with a new set of reducers. Looks like the majority of the page is the table. My current proposal for reusing TaskTable is:

In order for TaskTable to be reusable, it needs to be decoupled with any particular state. Kind of problematic to do it right now since a bunch of things in queue rely on the current implementation. Easiest proposal is to 

1. add another export in TaskTable that exports the component without bindings. 
2. add a function in task table props that allows for declaration of columns outside of TaskTable.jsx. Maybe take in something like
```
additionalColumns: [
  {
     header: 
     span:
     valueFunction: () => {}
     getSortValue:   },
  {
     header: 
     span:
     valueFunction: () => {}
     getSortValue   }...
]
```
3. If additionalColumns is present, properties within it would also get compacted into getQueueColumns. Can also just add a switch on the current columns compacted vs new additionalColumns param if current code execution would be problematic with the props being extracted out.

I think this approach would leave keep queue functionality while allowing custom modifications of the TaskTable. In the future we probably want to extract out the current column logic into separate functions that are imported into each TaskTable as needed.

**Comment by mdbenjam** [comment]
> In the future we probably want to extract out the current column logic into separate functions that are imported into each TaskTable as needed.

@lowellrex and I have talked about this and are very much in favor of it! If you want to just make that change I think that'd be 👍 . I don't think it would be a large lift, and would just keep things cleaner. Otherwise, the plan you've laid out with having some of our own functions passed in, and some of the current column-props passed in seems fine.

**Comment by lowellrex** [comment]
Here is a ticket I whipped up with general thoughts on [how to refactor the `TaskTable` component](https://github.com/department-of-veterans-affairs/caseflow/issues/7863). The second bullet there specifically calls out passing columns into the component (and is maybe the only refactor in that ticket worth doing in the near term).

**Pull request closed without merging**